src/main/java/com/suse/saltstack/netapi/event/EventStream.java

```diff
@@ -183,30 +199,57 @@ public void onOpen(Session session, EndpointConfig config) throws IOException {
     }
 
     /**
-     * Notify listeners on each event received on the WebSocket.
+     * Notify listeners on each event received on the websocket and buffer partial messages.
      *
-     * @param message The message received on this WebSocket
+     * @param partialMessage partial message received on this websocket
+     * @param last indicate the last part of a message
+     * @throws MessageTooBigException in case the message is longer than maxMessageLength
      */
     @OnMessage
-    public void onMessage(String message) {
-        if (message != null && !message.equals("server received message")) {
-            // Salt API adds a "data: " prefix that we need to ignore
-            Event event = JsonParser.EVENTS.parse(message.substring(6));
-            synchronized (listeners) {
-                listeners.stream().forEach(l -> l.notify(event));
+    public void onMessage(String partialMessage, boolean last)
+            throws MessageTooBigException {
+        if (maxMessageLength > 0 &&
+                messageBuffer.length() + partialMessage.length() > maxMessageLength) {
```

Comment:
I understand I suggested this construct, but since this is external data and assuming "bad things" happen, a better way to write this that avoids possibility of integer overflow (which could bypass this check) would be,

    partialMessage.length() > maxMessageLength - messageBuffer.length()

The only difference here should be checked exception vs. unchecked exception from messageBuffer in case of "bad things".